### PR TITLE
Add aten reshape

### PR DESCRIPTION
**Pull request by bszmelcz**
#### Details:
- [x] add `aten::reshape` operator
- [x] add layer test for `aten::reshape` operator

#### Ticket:
94262

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_reshape.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        ref_net = None
+
+        return aten_reshape(shape), ref_net, "aten::reshape"
+
+    @pytest.mark.parametrize(("shape"), [
+        [2, 3],
+        [3, 2],
+        [3, 2],
+        [6, 1],
+        [1, 6],
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_reshape.py]: Please add case with -1 and also please change the test from scalar of size 6 reshape to at least 2d tensor reshaping.

**Comment by mvafin** [comment]
I see that reshape operation is very similar to view, so it is most likely needs same handling if `prim::ListConstruct` is at second input of this operation. I recommend checking models that contain reshape and if this is really the case, move the replacing of `prim::ListConstruct` done in `aten::view` translator to a transformation stage. And have same translator for view and reshape.

File: src/frontends/pytorch/src/op/reshape.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_reshape(NodeContext& context) {
+    auto shape_node = context.get_input(1).get_node();
+    auto shape_node_fw_node = dynamic_cast<PtFrameworkNode*>(shape_node);
+    std::shared_ptr<ov::Node> reshape;
+    // TODO: move this to transform stage
+    if (shape_node_fw_node && shape_node_fw_node->get_decoder()->get_op_type() == "prim::ListConstruct") {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/reshape.cpp:20]: I was thinking you will move this to transformation step, not copy from `aten::view`

**Comment by slyalin** [comment]
@bszmelcz, can we go forward with this? If new changes are done to cover all feedback gathered, please retrigger review request to @mvafin.

File: src/frontends/pytorch/src/op/reshape.cpp (hunk 2 above)
Comment (bszmelcz) [reply src/frontends/pytorch/src/op/reshape.cpp:20]: As we discussed I will add this transformation as soon as other required operations will be ready.

**Review by mvafin — APPROVED** [review]
@bszmelcz Please create ticket to move `ListConstruct` handling to transformation step.

**Comment by bszmelcz** [comment]
> @bszmelcz Please create ticket to move `ListConstruct` handling to transformation step.

@mvafin It is already created :)

**Pull request merged**